test/e2e/flowaggregator_test.go

```diff
@@ -1815,6 +1837,35 @@ func createToExternalTestServer(t *testing.T, data *TestData) *PodIPs {
 	return serverIPs
 }
 
+func getAndCheckFlowAggregatorMetrics(t *testing.T, data *TestData) error {
+	flowAggPod, err := data.getFlowAggregator()
+	if err != nil {
+		return fmt.Errorf("error when getting flow-aggregator Pod: %v", err)
+	}
+	podName := flowAggPod.Name
+	command := []string{"antctl", "get", "recordmetrics", "-o", "json"}
+	if err := wait.Poll(defaultInterval, 2*defaultTimeout, func() (bool, error) {
+		stdout, _, err := runAntctl(podName, command, data)
+		if err != nil {
+			t.Logf("error when requesting recordmetrics, %v", err)
+			return false, nil
+		}
+		metrics := &recordmetrics.Response{}
+		if err := json.Unmarshal([]byte(stdout), metrics); err != nil {
+			t.Fatalf("error when decoding recordmetrics, %v", err)
+			return false, nil
```

Comment:
probably better to just return the error here:
````go
return false, fmt.Errorf("error when decoding recordmetrics: %w", err)
```

Reply:
Yes. We should just return error here. Thanks.